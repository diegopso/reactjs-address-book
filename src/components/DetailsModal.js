import React from 'react'
import { Modal, Layout, Row, Col, Image, Typography } from 'antd'
import { upperCaseFirst } from '../helpers/string'
import { EnvironmentOutlined, PhoneOutlined, WhatsAppOutlined } from '@ant-design/icons'

const { Text, Title } = Typography

const DetailsModal = ({ user, visible, onDismiss }) => {
  if (!user.location) {
    return <></>
  }

  return (
    <Modal
      className="user-details"
      title="User Details"
      visible={visible}
      onCancel={onDismiss}
      footer={null}>
      <Layout>
        <Row>
          <Col span={8}>
            <Image src={user.picture.large} width={'80%'} preview={false} />
          </Col>
          <Col span={12}>
            <Row>
              <Col span={24}>
                <Title level={2}>{upperCaseFirst(user.name.first) + ' ' + upperCaseFirst(user.name.last)}</Title>
              </Col>
              <Col span={2}>
                <PhoneOutlined />
              </Col>
              <Col span={22}>
                <Text>{user.phone}</Text>
              </Col>
              <Col span={2}>
                <WhatsAppOutlined />
              </Col>
              <Col span={22}>
                <Text>{user.cell}</Text>
              </Col>
              <Col span={2}>
                <EnvironmentOutlined />
              </Col>
              <Col span={22}>
                <Text>{user.location.street.name} {user.location.street.number}</Text><br />
                <Text>{user.location.postcode} {user.location.city}</Text><br />
                <Text>{user.location.state}, {user.location.country}</Text>
              </Col>
            </Row>

          </Col>
        </Row>
      </Layout>
    </Modal>
  )
}

export default DetailsModal
