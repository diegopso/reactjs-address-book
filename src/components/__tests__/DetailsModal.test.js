import React from 'react'
import { describe, it, expect } from '@jest/globals'
import { shallow } from 'enzyme'
import DetailsModal from '../DetailsModal'

describe('Testing DetailsModal component:', () => {
  const userMock = {
    name: {
      first: 'brad',
      last: 'gibson'
    },
    picture: {
      large: 'https://randomuser.me/api/portraits/thumb/men/75.jpg'
    },
    location: {
      street: {
        number: 9278,
        name: 'New road'
      },
      city: 'kilcoole',
      state: 'waterford',
      postcode: '93027',
      country: 'Spain'
    }
  }

  it('Renders correctly.', () => {
    const wrapper = shallow(<DetailsModal user={userMock} />)
    expect(wrapper).toMatchSnapshot()
  })
})
