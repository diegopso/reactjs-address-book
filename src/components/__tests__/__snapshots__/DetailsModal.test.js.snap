// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DetailsModal component: Renders correctly. 1`] = `
<Modal
  className="user-details"
  confirmLoading={false}
  footer={null}
  okType="primary"
  title="User Details"
  visible={false}
  width={520}
>
  <Layout>
    <Row>
      <Col
        span={8}
      >
        <Image
          preview={false}
          src="https://randomuser.me/api/portraits/thumb/men/75.jpg"
          width="80%"
        />
      </Col>
      <Col
        span={12}
      >
        <Row>
          <Col
            span={24}
          >
            <Title
              level={2}
            >
              Brad Gibson
            </Title>
          </Col>
          <Col
            span={2}
          >
            <ForwardRef(PhoneOutlined) />
          </Col>
          <Col
            span={22}
          >
            <Text />
          </Col>
          <Col
            span={2}
          >
            <ForwardRef(WhatsAppOutlined) />
          </Col>
          <Col
            span={22}
          >
            <Text />
          </Col>
          <Col
            span={2}
          >
            <ForwardRef(EnvironmentOutlined) />
          </Col>
          <Col
            span={22}
          >
            <Text>
              New road
               
              9278
            </Text>
            <br />
            <Text>
              93027
               
              kilcoole
            </Text>
            <br />
            <Text>
              waterford
              , 
              Spain
            </Text>
          </Col>
        </Row>
      </Col>
    </Row>
  </Layout>
</Modal>
`;
